pydantic/_internal/_generate_schema.py

```diff
@@ -210,6 +211,42 @@ def modify_model_json_schema(
     return json_schema
 
 
+JsonEncoders = Dict[Type[Any], JsonEncoder]
+
+
+def _add_custom_serialization_from_json_encoders(
+    json_encoders: JsonEncoders | None, tp: Any, schema: CoreSchema
+) -> CoreSchema:
+    """Iterate over the json_encoders and add the first matching encoder to the schema.
+
+    Args:
+        json_encoders (JsonEncoders | None): A dictionary of types and their encoder functions.
+        tp (Any): The type to check for a matching encoder.
+        schema (CoreSchema): The schema to add the encoder to.
+    """
+    if json_encoders is None:
```

Comment:
```suggestion
    if not json_encoders:
```
also handles the empty-dict case 🤷‍♂️